cmake/ystdlib-helpers.cmake

```
@@ -0,0 +1,231 @@
+    if(IS_INTERFACE_LIB)
+        if(ARG_PRIVATE_LINK_LIBRARIES)
+            message(
+                FATAL_ERROR
+                "`PRIVATE_LINK_LIBRARIES` disabled for header-only library ${ALIAS_TARGET_NAME}."
+            )
+        endif()
+        add_library(${ARG_NAME} INTERFACE)
+        target_link_libraries(${ARG_NAME} INTERFACE ${ARG_PUBLIC_LINK_LIBRARIES})
+
+        target_compile_features(${ARG_NAME} INTERFACE cxx_std_20)
+    else()
+        add_library(${ARG_NAME})
+        target_sources(${ARG_NAME} PRIVATE ${ARG_PRIVATE_SOURCES})
+        target_link_libraries(
+            ${ARG_NAME}
+            PUBLIC
+                ${ARG_PUBLIC_LINK_LIBRARIES}
+            PRIVATE
+                ${ARG_PRIVATE_LINK_LIBRARIES}
+        )
+        target_compile_features(${ARG_NAME} PUBLIC cxx_std_20)
+    endif()
+
+    add_library(${ALIAS_TARGET_NAME} ALIAS ${ARG_NAME})
+
+    target_sources(
+        ${ARG_NAME}
+        PUBLIC
+            FILE_SET HEADERS
+            BASE_DIRS
+                "$<BUILD_INTERFACE:${ARG_BUILD_INCLUDE_DIRS}>"
+                "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+            FILES ${ARG_PUBLIC_HEADERS}
+    )
+endfunction()
+
+# Adds a C++ 20 test executable named `unit-test-NAME` that will be built with `SOURCES` and linked
+# with `LINK_LIBRARIES`, in addition to Catch2.
+#
+# @param {string} NAME
+# @param {string} NAMESPACE
+# @param {string[]} SOURCES
+# @param {string[]} [LINK_LIBRARIES]
+# @param {string} [UNIFIED_TEST_TARGET] If set, `SOURCES` and `LINK_LIBRARIES` are also added to
+# `UNIFIED_TEST_TARGET`.
+function(add_catch2_tests)
+    set(SINGLE_VALUE_ARGS
+        NAME
+        NAMESPACE
+        UNIFIED_TEST_TARGET
+    )
+    set(MULTI_VALUE_ARGS
+        SOURCES
+        LINK_LIBRARIES
+    )
+    set(REQUIRED_ARGS
+        NAME
+        NAMESPACE
+        SOURCES
+    )
+    cmake_parse_arguments(ARG "" "${SINGLE_VALUE_ARGS}" "${MULTI_VALUE_ARGS}" ${ARGN})
+    check_required_arguments_exist("${REQUIRED_ARGS}")
+
+    set(ALIAS_TARGET "${ARG_NAMESPACE}::${ARG_NAME}")
+    set(UNIT_TEST_TARGET "unit-test-${ARG_NAME}")
+
+    add_executable(${UNIT_TEST_TARGET})
+    target_sources(${UNIT_TEST_TARGET} PRIVATE ${ARG_SOURCES})
+    target_link_libraries(
+        ${UNIT_TEST_TARGET}
+        PRIVATE
+            Catch2::Catch2WithMain
+            ${ARG_LINK_LIBRARIES}
+            ${ALIAS_TARGET}
+    )
+    target_compile_features(${UNIT_TEST_TARGET} PRIVATE cxx_std_20)
+    set_property(
+        TARGET
+            ${UNIT_TEST_TARGET}
+        PROPERTY
+            RUNTIME_OUTPUT_DIRECTORY
+                ${CMAKE_BINARY_DIR}/testbin
+    )
+
+    if(ARG_UNIFIED_TEST_TARGET)
+        target_sources(${ARG_UNIFIED_TEST_TARGET} PRIVATE ${ARG_SOURCES})
+        target_link_libraries(
+            ${ARG_UNIFIED_TEST_TARGET}
+            PRIVATE
+                ${ARG_LINK_LIBRARIES}
+                ${ALIAS_TARGET}
```

```suggestion
                ${ALIAS_TARGET}
                ${ARG_LINK_LIBRARIES}
```

To alphabetize (my comment in the previous review was wrong, sorry).